### PR TITLE
Parsing of Csv files + tests with AssertJ (Max & Ghis)

### PR DESCRIPTION
**Pull request by Ghis26**
works for the 6 files

### CODE REVIEW

**Comment by Noksit** [comment]
+1

File: java/test/test/CSVFileTest.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    @Test
+    void shouldIgnoreBlanks() {
+        CSVFile csv = new CSVFile();
+
+        List<Country> result = csv.readCsv("data/country3.csv");
+
+        assertThat(result.get(1).getCode()).isNotEqualTo("  BE");
+        assertThat(result.get(2).getId()).isNotEqualTo("3 ");
```
[the remark this answers is not in the document]
Comment (Ghis26) [reply java/test/test/CSVFileTest.java]: thanks for your comment

File: java/test/test/CSVFileTest.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    @Test
+    void shouldAddIdIfEmpty() {
+        CSVFile csv = new CSVFile();
+
+        List<Country> result = csv.readCsv("data/country5.csv");
+
+        for (int i =0; i< result.size(); i++) {
+            assertThat(result.get(i).getId()).isEqualTo(i+1);
```
[the remark this answers is not in the document]
Comment (Ghis26) [reply java/test/test/CSVFileTest.java]: thanks for your comment

File: java/test/test/CSVFileTest.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        assertThat(result.get(2).getId()).isEqualTo(3);
+    }
+
+    @Test
+    void shouldIgnoreEmptyLines() {
+        CSVFile csv = new CSVFile();
+
+        List<Country> result = csv.readCsv("data/country2.csv");
+
+        assertThat(result.size()).isEqualTo(3);
```
[the remark this answers is not in the document]
Comment (Ghis26) [reply java/test/test/CSVFileTest.java]: I didn't know this method.

**Comment by mbaechler** [comment]
Could you please squash your commits ?

File: java/java.iml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mbaechler) [inline java/java.iml]: please remove this kind of file from your commit

File: java/src/main/company/CSVFile.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+        List<Country> result = new ArrayList<>();
+        FileReader fr;
+        BufferedReader br;
+        String[] str;
+
+        int id = 1;
+
+        try {
+            fr = new FileReader(file);
```
Comment (mbaechler) [inline java/src/main/company/CSVFile.java]: you are probably leaking the FileReader

File: java/test/test/CSVFileTest.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+import static org.assertj.core.api.Assertions.*;
+
+
+
+class CSVFileTest extends CSVFile {
+
+
+    @Test
+    void readFile() {
```
Comment (mbaechler) [inline java/test/test/CSVFileTest.java]: in the nominal case, you should probably assert that every fields are well parsed, don't you think ?

File: java/test/test/CSVFileTest.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+        CSVFile csv = new CSVFile();
+
+        List<Country> result = csv.readCsv("data/country4.csv");
+
+        assertThat(result.get(0).getId()).isNotEqualTo("Id");
+        assertThat(result.size()).isEqualTo(3);
+    }
+
+    @Test
+    void shouldAddIdIfEmpty() {
```
Comment (mbaechler) [inline java/test/test/CSVFileTest.java]: what make you think that Id is an auto-incremented value ?

File: java/untitled104.iml [hunk 8]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mbaechler) [inline java/untitled104.iml]: please don't commit this kind of file

File: java/test/test/CSVFileTest.java (hunk 7 above)
Comment (Ghis26) [reply java/test/test/CSVFileTest.java]: In our opinion, every line (every Country object by the way)  must thave an Id . Lines on the file don't have any Id, we judged relevant to increment Id for each line.

File: java/test/test/CSVFileTest.java (hunk 6 above)
Comment (Ghis26) [reply java/test/test/CSVFileTest.java]: Effectively, it's the subject of the test. We verify if the id, name and code is well parsed in different lines. but the name of the test isn't relevant. Our apologies

Comment (mbaechler) [reply java/test/test/CSVFileTest.java]: please don't apology, that's not what review is for.
but every comment should lead to either a fix or an agreement on the fact it doesn't require a fix

File: java/test/test/CSVFileTest.java (hunk 7 above)
Comment (mbaechler) [reply java/test/test/CSVFileTest.java]: please drop this behaviour

File: java/src/main/company/CSVFile.java (hunk 5 above)
Comment (Ghis26) [reply java/src/main/company/CSVFile.java]: Done. The FileReader has been closed when the method is finished.

File: java/test/test/CSVFileTest.java (hunk 7 above)
Comment (Ghis26) [reply java/test/test/CSVFileTest.java]: Done with the last commit

File: java/test/test/CSVFileTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (mbaechler) [reply java/test/test/CSVFileTest.java]: Why didn't you changed your code to make us of hasSize ?

File: java/test/test/CSVFileTest.java [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    @Test
+    void shouldIgnoreIdFieldIfIsEmpty() {
+        CSVFile csv = new CSVFile();
+
+        List<Country> result = csv.readCsv("data/country5.csv");
+
+        for (int i = 0; i < result.size(); i++) {
+            assertThat(result.get(i).getId()).isEqualTo(i + 1);
```
Comment (mbaechler) [inline java/test/test/CSVFileTest.java]: are you sure that the assertions is what the test name describe ?

Comment (Ghis26) [reply java/test/test/CSVFileTest.java]: Ok, changes done

File: java/test/test/CSVFileTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (Ghis26) [reply java/test/test/CSVFileTest.java]: Done with the last commit

**Comment by mbaechler** [comment]
Good work

**Pull request closed without merging**